src/controller/embeddingController.ts

```diff
@@ -0,0 +1,169 @@
+import { eq } from 'drizzle-orm'
+import { t } from 'elysia'
+import { db } from '../db'
+import { questions } from '../db/schema'
+import {
+    generateQuestionAnswerEmbeddings
+} from '../services/embeddingService'
+import { embeddingRequestSchema } from '../types/embeddings'
+import { HttpStatusCode } from '../utils'
+import {
+    EmbeddingError
+} from '../utils/error/customEmbeddingError'
+
+/**
+ * Helper function to format error responses consistently
+ */
+const formatErrorResponse = (error: Error, defaultCode: string = 'UNKNOWN_ERROR', defaultStatusCode: HttpStatusCode = HttpStatusCode.INTERNALSERVERERROR, defaultRetryable: boolean = false) => {
```

Comment:
shouldn't the service be responsible for giving you the error,so you don't have to handle it here?